presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -185,7 +185,8 @@ private ColumnChunk readStruct(GroupField field)
             }
         }
         BooleanList structIsNull = ParquetStructColumnReader.calculateStructOffsets(field, columnChunk.getDefinitionLevels(), columnChunk.getRepetitionLevels());
-        Block rowBlock = RowBlock.fromFieldBlocks(structIsNull.toBooleanArray(), blocks);
+        boolean[] structIsNullVector = structIsNull.toBooleanArray();
+        Block rowBlock = RowBlock.fromFieldBlocks(structIsNullVector.length, Optional.ofNullable(structIsNullVector), blocks);
```

Comment:
`structIsNullVector` won't be null in this case so you only need `Optional.of`. Same for most of the uses.